# Module 2: Data Analytics with Python - Statistics

## Lab 2.2: Advance Statistics

<br><br><br><br><br><br>
## Objevtive
***
- ANOVA(Analysis Of Varience Test)

### **ANOVA (Analysis Of Variance) F-test**
***

This test works well when dealing with two groups, but sometimes we want to compare more than two groups at the same time. 
The Analysis of Variance (ANOVA) is a statistical inference test that lets you compare multiple groups at the same time.
F = Variability between groups or Variability within groups

Unlike the z and t-distributions, the F-distribution does not have any negative values because between and within-group variability are always positive due to squaring each deviation.

**One Way F-test (Anova)**

It tells whether two or more groups are similar or not based on their mean similarity and f-score.

p> 0.05 implies they are similar

p< 0.05 implies they are not similar


In [ ]:
%matplotlib inline
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols
url = "https://raw.githubusercontent.com/bluedataconsulting/AIMasteryProgram/main/Lab_Exercises/Module2/PlantGrowth.csv"
data = pd.read_csv(url)
mod = ols('weight ~ group', data=data).fit()      #we use ols to set up model
aov_table = sm.stats.anova_lm(mod,type=2)

print(aov_table)

            df    sum_sq   mean_sq         F   PR(>F)
group      2.0   3.76634  1.883170  4.846088  0.01591
Residual  27.0  10.49209  0.388596       NaN      NaN


**Conclusion: From AOV_Table we observe that value of p=0.01591 > 0.05 and therefore both groups are similar.**

### Thank You !!!